app/scripts/lib/operation-safener.ts

```diff
@@ -0,0 +1,137 @@
+import { type DebounceSettings, type DebouncedFunc, debounce } from 'lodash';
+import log from 'loglevel';
+
+export type { DebounceSettings } from 'lodash';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type AnyParams = any[];
+
+/**
+ * A type representing a generic operation that can be executed.
+ */
+export type Op = (...params: AnyParams) => unknown | Promise<unknown>;
```

Comment:
The deep generics here ended up being unnecessary because I decided to not return and values from the `Op` function. Kept them around anyway. I don't think do much harm, but I won't fight to keep them if someone else hates them enough to block this PR because of it. :-)